elks/fs/exec.c

```diff
@@ -352,6 +352,27 @@ static int FARPROC execve_aout(struct inode *inode, struct file *filp,
     /*
      *      Looks good. Get the memory we need
      */
+
+#if defined(CONFIG_ROMFS_FS) && defined(CONFIG_ROMFS_FS_DIRECT_CSEG)
+    if (filp->f_inode && filp->f_inode->i_sb
+        && filp->f_inode->i_sb->s_type->type == FST_ROMFS
+        && !(filp->f_pos & 15)
+        && mh.hlen == EXEC_MINIX_HDR_SIZE
+    ) {
+        // Point the code segment directly to in-memory ROMFS.
+        seg_code = seg_alloc_fixed(
+            filp->f_inode->u.romfs.seg + (filp->f_pos >> 4),
+            bytes_to_paras(mh.tseg),
+            SEG_FLAG_CSEG
+        );
```

Comment:
Please reformat as many arguments as will fit on an 80-90 char line, as this is the way the rest of the kernel is formatted (or at least is supposed to be).